lib/SymbolGraphGen/DeclarationFragmentPrinter.cpp

```diff
@@ -189,9 +189,17 @@ void DeclarationFragmentPrinter::printAbridgedType(const GenericTypeDecl *TD,
       case DeclKind::Protocol:
         printText(getTokenText(tok::kw_protocol));
         break;
-      case DeclKind::Class:
-        printText(getTokenText(tok::kw_class));
-        break;
+        case DeclKind::Class: {
+            auto *newthing = cast<ClassDecl>(TD);
+            if (newthing->isActor()) {
+                printText("actor");
```

Comment:
I do not know if `kw_action` exists. But if not, I suggest we should add it first and use the constant instead of using a hard cord string here.

```suggestion
                printText(tok::kw_actor);
```

Reply:
It looks like [`TokenKinds.def`](https://github.com/apple/swift/blob/main/include/swift/AST/TokenKinds.def) doesn't define an `actor` keyword, so unfortunately we'll need to spell it out. 😞

Reply:
That definition would go in [`TokenKinds.def`](https://github.com/apple/swift/blob/main/include/swift/AST/TokenKinds.def), but i would want to make extra sure that we can add the keyword there before doing it. I'm not sure who to ask, though.

Reply:
I've added a keyword declaration in for "actor", but I'm hesitant to do so:
1 - The token wasn't added earlier, but clearly the parser understands the keyword `actor` already, and it's not clear why it wasn't added back when `actor` first became a thing.
2 - I don't seem to be able to do the same with 'distributed', the modifier for 'actor', which makes the remaining piece asserting distributed actor or not sort of awkward.
3 - I'm uncertain of the implications of adding a token to the keyword list, and where else that ripples into. The parser at least. Looking through the parser, "actor" and "distributed actor" are handled are raw strings, not using `Tok::` structures.

A full build locally succeeded with the `tok::actor` added in there, as did the lit.py tests. (well, 3 failed, but the same three as the first run at this, and seemingly unrelated):

```
Failed Tests (3):
  Swift(macosx-arm64) :: Reflection/capture_descriptors.sil
  Swift(macosx-arm64) :: stdlib/CharacterRecognizer.swift
  Swift(macosx-arm64) :: stdlib/StringIndex.swift
```

Reply:
I added it in, just to try - but I can very easily pull it back out, and as I mentioned, I'm hesitant to go here because it hasn't been added previously, with the parser seeming to use the older keyword sets, but not exclusively.

Reply:
After creating the test and trying to verify it, I was having all sorts of trouble compiling everything, so I backed out that specific change and reverted to printing the raw string rather than using a token.

Reply:
Hmm, I don't know about the implications of adding the keyword there, maybe @ahoppen could advise?

Reply:
The parser handles `actor` and `distributed` as strings and not keywords, because they're not keywords. For example, the following is valid:

```
struct actor {}
var x: actor = actor()
```

Reply:
`actor` and `distributed` are contextual keywords and thus shouldn’t be in TokenKinds.def. Notice how `await`, `async` and all sorts of other keywords that were introduced after Swift 2 or 3 also aren’t in TokenKinds.def.

Reply:
Ah I see, thanks for chiming in! :-)

Reply:
Thank you!